### PR TITLE
feat(slack): improve slashcommand handling

### PR DESCRIPTION
**Pull request by Charles546**
* customizable `slash_notify`
 * support detailed help for individual commands
 * granular control on error notifications

### CODE REVIEW

File: slack/workflows.yaml [hunk 1]
```diff
@@ -138,41 +166,83 @@ workflows:
...
+                No detailed information available for this command.
+                {{- end }}
+              type: mrkdwn
+
+  slashcommand/respond:
     meta:
       description:
         - >
-          This workflow sends a announcement message to the channels listed in :code:`slash_notify`.  Used internally.
+          This workflow sends a respond message to the channels listed in :code:`slash_notify`.  Used internally.
```
Comment (wyardley) [inline slack/workflows.yaml:192]: response?

File: slack/workflows.yaml [hunk 2]
```diff
@@ -138,41 +166,83 @@ workflows:
...
+          message_type: $ctx.message_type,"normal"
+          message: $?ctx.notification
 
-  slashcommand/status:
+  slashcommand/announcement:
     meta:
       description:
         - >
-          This workflow sends a status message to the channels listed in :code:`slash_notify`.  Used internally.
+          This workflow sends a announcement message to the channels listed in :code:`slash_notify`.  Used internally.
```
Comment (wyardley) [inline slack/workflows.yaml:215]: 'an announcement'

**Review by wyardley — APPROVED** [review]
Does this help support subcommands as well?
Overall, lgtm. couple minor typos mentioned in the comments inline.

**Comment by Charles546** [comment]
Wait until `2.1.1` is deployed.

File: slack/workflows.yaml [hunk 3]
```diff
@@ -161,18 +162,26 @@ workflows:
...
+      - call_workflow: channel_translate
+        with:
+          channel_names: $?ctx.slash_notify
+        export:
+          notify: $?ctx.slash_notify
+      - unless_match:
+          - channel_ids: $ctx.channel_id
+          - slash_notify: $ctx.channel_fullname
+        export:
+          notify: $?ctx.notify
```
Comment (Charles546) [inline slack/workflows.yaml:243]: This line should be removed after `2.1.1`.

File: workflow_helper.yaml [hunk 4]
```diff
@@ -49,6 +49,7 @@ workflows:
             - "error"
             - "failure"
         export:
+          notify: $?ctx.notify
```
Comment (Charles546) [inline workflow_helper.yaml:52]: To be removed after `2.1.1`

**Pull request merged**